src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -234,6 +234,18 @@ public ITelemetryChannel TelemetryChannel
         /// </remarks>
         public IApplicationIdProvider ApplicationIdProvider { get; set; }
 
+        /// <summary>
+        /// Gets or sets a collection of string indicating if an experimental feature should be enabled.
+        /// The presence of a string in this collection will be evaluated as 'true'.
+        /// </summary>
+        /// <remarks>
+        /// This property allows the dev team to ship and evaluate features before adding these to the public API.
+        /// We are not committing to support any features enabled through this property.
+        /// Use this at your own risk.
+        /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public IEnumerable<string> ExperimentalFeatures { get; set; }
```

Comment:
where is this instantiated?

Reply:
new TC().ExperimentatlFeatures would throw now.

Reply:
This would have to be manually instantiated, either in config file or within code.

I could change this to `public IEnumerable<string> ExperimentalFeatures { get; set; } = new string[0]` if your concern is preventing it from throwing exceptions. But I don't see the value in instantiating an empty array that would never be used.

Please let me know if I am missing something.

Reply:
There is a null check in the other class